blockchain-albatross/src/history_store/history_store.rs

```diff
@@ -634,25 +742,59 @@ impl HistoryStore {
             }
         };
 
-        let mut leaf_hashes = vec![];
+        let mut tx_hashes = vec![];
 
-        // Seek to the first leaf hash at the given block number.
-        let mut cursor = txn.cursor(&self.block_db);
-        let leaf_hash = match cursor.seek_key::<u32, Blake2bHash>(&block_number) {
-            None => return leaf_hashes,
-            Some(v) => v,
-        };
+        // Seek to the first transaction hash at the given address.
+        let mut cursor = txn.cursor(&self.address_db);
 
-        leaf_hashes.push(leaf_hash);
+        cursor.seek_key::<Address, OrderedHash>(address);
 
-        loop {
-            // Get next leaf hash.
-            match cursor.next_duplicate::<u32, Blake2bHash>() {
-                Some((_, leaf_hash)) => leaf_hashes.push(leaf_hash),
+        // Then go to the last transaction hash at the given address.
+        match cursor.last_duplicate::<OrderedHash>() {
+            Some(v) => tx_hashes.push(v.hash),
+            None => return tx_hashes,
+        };
+
+        while tx_hashes.len() < max as usize {
+            // Get previous transaction hash.
+            match cursor.prev_duplicate::<Address, OrderedHash>() {
+                Some((_, v)) => tx_hashes.push(v.hash),
                 None => break,
             };
         }
 
-        leaf_hashes
+        tx_hashes
+    }
+
+    /// Returns the number of transactions (no inherents) associated to the given address.
+    pub fn get_num_txs_by_address(
```

Comment:
Naming: this function does not necessarily return the number of stored txs. If an entry was deleted that was not the most recent tx, there's a gap in the indices and the number of txs is less than the last index + 1. Instead, I suggest to call this method `get_next_tx_index_for_address` or `get_last_tx_index_for_address` (and remove the +1) or something in that direction.

Reply:
You're right. Good catch. In that case I'll also make the method private, since it doesn't have any use outside the History Store